File: distance.go
package main
 
import (
    "fmt"
    "math"
)
 
type location struct {
    lat float64 // latitude, radians
    long float64 // longitude, radians
}
 
func deg2rad(lat, long float64) location {
    return location{lat * math.Pi / 180, long * math.Pi / 180}
}
 
const R = 3963.1676 // Radius of earth in miles
 
func distance(p1, p2 location) float64 {
    var dlat float64 = p1.lat - p2.lat
    var dlong float64 = p1.long - p2.long

    var a float64 = math.Pow(math.Sin(dlat/2), 2) + math.Cos(p1.lat)*math.Cos(p2.lat)*math.Pow(math.Sin(dlong/2), 2)
    return 2 * R * math.Atan2(math.Sqrt(a), math.Sqrt(1 - a))
}

func main() {
    fmt.Println(distance(deg2rad(37.78, -122.43), deg2rad(37.78, -122.39)))
}

File: distance.py
#!/usr/bin/env python2.7
# -*- coding: utf-8 -*-

import numpy as np

sin = np.sin
cos = np.cos
atan2 = np.arctan2
sqrt = np.sqrt
pi = np.pi

R = 3963.1676  # Radius of earth in miles


def distance(lat1, long1, lat2, long2):
    """
    Calculate the Haversine distance in miles between two locations based
    on their latitudes and longitudes

    Inputs
    ------

    lat1 : float
        latitude in degrees of first point
    long1 : float
        longitude in degrees of first point
    lat2: float
        latitude in degrees of second point
    long2 : float
        longitude in degrees of second point

    """
    lat1 = lat1 * pi / 180.  # convert to radians
    lat2 = lat2 * pi / 180.  # convert radians
    dlat = (lat1 - lat2)
    dlong = (long1 - long2) * pi / 180.  # difference in radians

    a = sin(dlat/2)**2 + cos(lat1) * cos(lat2) * sin(dlong/2)**2
    return 2 * R * atan2(sqrt(a), sqrt(1-a))
